quickcheck-dynamic/src/Test/QuickCheck/DynamicLogic/CanGenerate.hs

```diff
@@ -8,14 +8,14 @@ import Test.QuickCheck
 --   otherwise @True@ (and we know such an x can be generated).
 canGenerate :: Double -> Gen a -> (a -> Bool) -> Bool
 canGenerate prob g p = unsafePerformIO $ tryToGenerate 1
- where
-  tryToGenerate luck
-    | luck < eps = return False
-    | otherwise = do
-        x <- generate g
-        if p x
-          then return True
-          else tryToGenerate (luck * (1 - prob))
+  where
```

Comment:
Because previously it wasn't indenting `where` correctly (in my opinion...).

In general I'm not super happy with the formatter. But perhaps poking around with it could be a separate PR.

Reply:
I am in the opinion that one strict standardised formatter without option is the way to go and I personally don't care as long as we are consistent.

Reply:
I wouldn't mind as long as that formatter wasn't producing poorly formatted code like:
```
[
        ( max 1 $ 10 - length (ctxAtType @ThreadId ctx)
        , return $ Some Spawn
        )
      ,
        ( 2 * Map.size (regs s)
        , Some <$> (Unregister <$> probablyRegistered s)
        )
      ,
        ( 10
        , Some <$> (WhereIs <$> probablyRegistered s)
        )
      ]
        ++ [ ( max 1 $ 3 - length (dead s)
             , Some <$> (KillThread <$> arbitraryVar ctx)
             )
           | not . null $ ctxAtType @ThreadId ctx
           ]
        ++ [ ( max 1 $ 10 - Map.size (regs s)
             , Some <$> (Register <$> probablyUnregistered s <*> arbitraryVar ctx)
             )
           | not . null $ ctxAtType @ThreadId ctx
           ]
```
I mean, why is the `++` indented?! why are there newlines after the commas in the first list and not in the second list?!

Or this gem:
```
nextState s (KillThread tid) _ =
    s
      { dead = tid : dead s
      , regs = Map.filter (/= tid) (regs s)
      }
```

I agree that having one standard and no options is a good thing. But the problem is when that standard is doing stuff that isn't very readable.

Likewise I don't like that function clauses that could have aligned `=` don't etc. but I can deal with that - but the stuff above I think is just wrong.

I can live with it if we don't have a better option though.

Reply:
fourmolu is a configurable ormolu, so I guess it should be possible to  something more to your likings. As I said, I don't care and will use whatever is there so feel free to tweak it :)

Reply:
Ok, then I'll see what I can do. Let's keep the current changes for now and I'll throw together a PR with a different tool / better configuration if I get to it on my TODO list. Not a super high priority at the moment.